### PR TITLE
GB-57: Create LibrariesPage layout

### PR DESCRIPTION
**Pull request by ksadoff**
## Jira Ticket Number
[GB-57](https://gaming-backlog.atlassian.net/browse/GB-57)

## Short description
This creates the basic structure for the page that displays the user's libraries, as well as basic routing and unit tests for the new page.

## Type of Change
- [x] New feature
- [ ] Bug fix
- [ ] Documentation
- [ ] Other (please specify)

### CODE REVIEW

File: src/api/libraryApi.ts [hunk 1]
```diff
@@ -1,19 +1,24 @@
...
-// TODO: will we eventually want this to be get all libraries for a single user? Will have to update backend endpoints
-export function getAllLibraries(): Promise<Library[]> {
-    return fetch(homeUrl + librariesBaseUrl, {
+export function getAllLibrariesWithGames(): Promise<Library[]> {
+    return fetch(homeUrl + librariesBaseUrl + "withGames/", {
         method: "GET",
     })
-    .then((response) => response.json());
+    .then((response) => {
+        console.log(response);
```
Comment (trauman413) [inline src/api/libraryApi.ts:10]: We should figure out how we want to do logging in the future. We could get a logger library if we want, not sure if Kibana is something we can get for free or cheap but maybe down the line? (i also vaguely hate kibana so not opposed to not using it lol)

File: src/pages/LibrariesPage.tsx [hunk 2]
```diff
@@ -1,27 +1,71 @@
...
+                <div>
+                    {(preview.games.map((game) => 
+                    <a>
+                        <p>{game.name}</p>
+                    </a>))}
+                </div>
+            )
+        } else {
+            const list = [];
+            for (let i = 0; i < 5; i++) {
```
Comment (trauman413) [inline src/pages/LibrariesPage.tsx:41]: Does it count as a magic number if the function is called "getFirstFIVEGames" lol.

**Review by trauman413 — APPROVED** [review]
left some small comments but i'm good approving as is

File: src/api/libraryApi.ts (hunk 1 above)
Comment (ksadoff) [reply src/api/libraryApi.ts:10]: Definitely on board with discussing this! Created a ticket to track it https://gaming-backlog.atlassian.net/browse/GB-61

File: src/pages/LibrariesPage.tsx (hunk 2 above)
Comment (ksadoff) [reply src/pages/LibrariesPage.tsx:41]: Agree about making this a bit more flexible in the future - was just basing this on the designs as we have them now!

File: src/api/libraryApi.ts (hunk 1 above)
Comment (ksadoff) [reply src/api/libraryApi.ts:10]: Oops also realizing I did not mean to leave that log in there! Will remove it in my next PR

**Pull request merged**